plugins/config/lib/configService.js

```diff
@@ -2524,37 +2524,52 @@ function ConfigService(context) {
       respondWithJsonError(response,"Cannot update a non-leaf resource",HTTP_STATUS_BAD_REQUEST,request.resourceURL);
       return 1;
     }
-    else if (!request.body) {
-      respondWithJsonError(response,"Could not access PUT body.",HTTP_STATUS_BAD_REQUEST,request.resourceURL);
-    }
-    else {
-      let b64 = request.query.b64;
-      let isB64 = b64 ? (b64.toLowerCase() == 'true') : false;
-      let timestamp = request.query.lastmod;
-      /* NOTE: here, scope only indicates where to place the files. aggregation policy is ignored */
-      if (!request.currentResourceList && itemName.length>0) {
-        //Replace or create one file
-        accessLogger.debug(`Configuration service handling element write request. Resource=${request.resourceURL}, Element=${itemName}, Scope=${request.scope}.`);
-        restCheckModifiedTimestamp(itemName,request.directories,request.scope,lastPath,timestamp).then(()=> {
-          replaceOrCreateFile(response, itemName, request.directories,request.scope,lastPath, request.resourceURL, request.body, request.body.length);
-        }, (err)=> {
-          if (err && err.message === 'Timestamp mismatch'){
-            logger.warn(`Could not delete resource due to timestamp mismatch. Resource=${request.resourceURL}, Element=${itemName}`);
-            respondWithJsonError(response,`Timestamp mismatch`,HTTP_STATUS_BAD_REQUEST,request.resourceURL);
-          } else {  
-            logger.warn(`Failed to check resource timestamp. Resource=${request.resourceURL}, Element=${itemName}, Err=${err.stack}`);
-            respondWithJsonError(response,`Timestamp check failure`,HTTP_STATUS_INTERNAL_SERVER_ERROR,request.resourceURL);
-          }
-        });
-      }
-      else {
-        //this also means deleting files that were previously there and not listed.
-        respondWithJsonError(response,"Response type not implemented.",HTTP_STATUS_NOT_IMPLEMENTED,request.resourceURL);
-        /*
-          replaceOrCreateDirectoryFiles(response,itemName,request.currentResourceObject,request.currentResourceList,request.directories,
-          request.scope,lastPath,request.resourceURL);
-        */
+
+    if (typeof request.body == 'string') {
+      try {
+        //We only support JSON storage for now.
+        //If we attempt to write out a string that isnt JSON, retrieval will be broken.
+        //This also handles the case in which body was just empty...
+        const bodyTest = JSON.parse(request.body);
```

Comment:
Style suggestion: the catch block for this is almost 40 lines away (and it would catch and report any exception as a JSON error), and the variable is never used, so it would be puzzling to figure out why this call is here. I'd improve the "locality" of this code (NOT TESTED):

```
    if (typeof request.body !== 'string') {
      respondWithJsonError(response,"Could not access PUT body.",HTTP_STATUS_BAD_REQUEST,request.resourceURL);
      return;
    }
    try {
      JSON.parse(request.body);
    } catch (e) {
      respondWithJsonError(response,"PUT body is not JSON.",HTTP_STATUS_BAD_REQUEST,request.resourceURL);
      return;
    }
    let b64 = request.query.b64;
    ...
  ```

Reply:
Thanks. I did this now.
The reason why bodyTest isn't used is that the body is written out into a file, so keeping it as a string (as it was sent) is good, but need to check that it's valid JSON first otherwise future GETs will throw exceptions either server-side or client-side.